zokrates_core/src/semantics.rs

```diff
@@ -470,8 +548,21 @@ impl<'ast, T: Field> Checker<'ast, T> {
                             .get(import.symbol_id)
                             .cloned();
 
-                        match (function_candidates.len(), type_candidate) {
-                            (0, Some(t)) => {
+                        // find constant definition candidate
+                        let const_candidate = state
+                            .typed_modules
+                            .get(&import.module_id)
+                            .unwrap()
+                            .constants
+                            .as_ref()
+                            .and_then(|tc| tc.get(import.symbol_id))
+                            .and_then(|sym| match sym {
+                                TypedConstantSymbol::Here(tc) => Some(tc),
+                                _ => None,
```

Comment:
`fold_constant_symbol` recursively finds `Here` so in the example where we import an imported constant, it does the recursion until we are in the module that actually defined the constant (`There` -> `There` -> `Here`). I guess the default arm should be  `unreachable`.

Reply:
Pretty much yes, https://github.com/Zokrates/ZoKrates/pull/792/commits/9f69ea37719a16ad8fc1887645514cb9ac5e31bb#diff-bd66e0ee29ac00ea22ffd0580bc6a2d61798d9c7a58c7a3c61f91f9a89a3a36fR655

Reply:
Right, that's where I think having this map be canonical and store c instead of Here(c) would remove a bunch of these unreachable cases.

Reply:
As we discussed, we can store only the Type of the constant here, because the value is never used in the semantic phase. This removes the unreachable cases.